styleguide/javascript/README.md

```diff
@@ -0,0 +1,140 @@
+# JavaScript Styleguide
+
+## Tooling
+
+#### Avoid jQuery
+
+Just don't use it. It's better to use standard APIs.
+
+[Should you use or learn jQuery in 2020?](https://flaviocopes.com/jquery/)
+
+#### Avoid lodash
+
+You probably don't need it. And if you do, import the functions you need like this:
+
+``` javascript
+import foo from 'lodash/foo'
+```
+
+Don't do this:
+
+``` javascript
+import { foo } from 'lodash'
+import _ from 'lodash'
+```
+
+The latter will make tree-shaking impossible.
+Read more about [lodash and tree-shaking](https://www.azavea.com/blog/2019/03/07/lessons-on-tree-shaking-lodash/)
+
+#### Use Prettier
+
+Use our [prettier-config](https://github.com/barsoom/prettier-config)
+
+#### Use eslint
+
+Use [eslint-config-barsoom](https://github.com/barsoom/eslint-config-barsoom)
+
+#### When using Babel
+
+Try to avoid using too many experimental features.
+
+Stage 3 is fine and stage 2 can be ok.
+
+Just don't go crazy with everything shiny and new that is likely to change, or maybe not even get into the standard.
+
+#### Prefer npm over yarn
+
+When Yarn was initially released, it solved a lot of problems that npm had.
+
+Since then, npm has improved a lot and nowadays there isn't really any reason to use yarn over npm.
+
+[Why Are You Still Using Yarn in 2018?](https://iamturns.com/yarn-vs-npm-2018/)
+
+#### Dependencies
+
+In the npm repository, you can find packages for almost anything.
+
+It might be tempting to add a dependency for something, but if it's simple enough it's probably better to implement it yourself. (Looking at you, [left-pad](https://qz.com/646467/how-one-programmer-broke-the-internet-by-deleting-a-tiny-piece-of-code/))
+
+## Code style
+
+#### Regular functions vs. arrow functions
+
+Use `function` for named functions and arrow functions for unnamed closures.
```

Comment:
It's nice when styleguides say *why*. Even if the reason why is just arbitrary consistency. In this case I guess it's about hoisting?

Reply:
It's mainly for readability. :)

Added the reason to the line below.